test/org/sagebionetworks/bridge/services/AuthenticationServiceTest.java

```diff
@@ -1,528 +0,0 @@
-        
-        AccountWorkflowService accountWorkflowServiceSpy = spy(accountWorkflowService);
-        authService.setAccountWorkflowService(accountWorkflowServiceSpy);
-
-        // Second sign up
-        authService.signUp(testUser.getStudy(), testUser.getStudyParticipant());
-        
-        ArgumentCaptor<AccountId> accountIdCaptor = ArgumentCaptor.forClass(AccountId.class);
-        verify(accountWorkflowServiceSpy).notifyAccountExists(eq(testUser.getStudy()), accountIdCaptor.capture());
-        assertEquals(testUser.getStudyIdentifier().getIdentifier(), accountIdCaptor.getValue().getStudyId());
-        assertEquals(testUser.getId(), accountIdCaptor.getValue().getId());
-    }
-    
-    @Test
-    public void resendEmailVerificationLooksSuccessfulWhenNoAccount() throws Exception {
-        AccountId accountId = AccountId.forEmail(TestConstants.TEST_STUDY_IDENTIFIER, "notarealaccount@sagebase.org");
-        authService.resendVerification(ChannelType.EMAIL, accountId);
-    }
-    
-    @Test
-    public void requestResetPasswordLooksSuccessfulWhenNoAccount() throws Exception {
-        SignIn signIn = new SignIn.Builder().withStudy(TEST_STUDY_IDENTIFIER).withEmail("notarealaccount@sagebase.org")
-                .build();
-        authService.requestResetPassword(study, false, signIn);
-    }
-    
-    // Consent statuses passed on to sessionInfo
-    
-    @Test
-    public void consentStatusesPresentInSession() throws Exception {
-        // User is consenting
-        testUser = helper.getBuilder(AuthenticationServiceTest.class)
-                .withConsent(true).withSignIn(true).build();
-
-        JsonNode info = UserSessionInfo.toJSON(testUser.getSession());
-        
-        TypeReference<Map<SubpopulationGuid,ConsentStatus>> tRef = new TypeReference<Map<SubpopulationGuid,ConsentStatus>>() {};
-        Map<SubpopulationGuid,ConsentStatus> statuses = BridgeObjectMapper.get().readValue(info.get("consentStatuses").toString(), tRef); 
-        
-        ConsentStatus status = statuses.get(SubpopulationGuid.create(testUser.getStudyIdentifier().getIdentifier()));
-        assertTrue(status.isConsented());
-        assertEquals(testUser.getStudyIdentifier().getIdentifier(), status.getSubpopulationGuid());
-    }
-    
-    @Test
-    public void existingLanguagePreferencesAreLoaded() {
-        List<String> LANGS = ImmutableList.of("en","es");
-        testUser = helper.getBuilder(AuthenticationServiceTest.class)
-                .withConsent(true).withLanguages(LANGS).withSignIn(true).build();
-        
-        authService.signOut(testUser.getSession());
-        
-        Study study = studyService.getStudy(testUser.getStudyIdentifier());
-        CriteriaContext context = testUser.getCriteriaContext();
-        
-        UserSession session = authService.signIn(study, context, testUser.getSignIn());
-        assertEquals(LANGS, session.getParticipant().getLanguages());
-    }
```

Comment:
Should these language tests be moved to unit tests?

Reply:
Added a unit test to verify that persisted language preferences are used over changed language preferences in the context object.